core/src/main/java/com/valtech/aem/saas/core/http/client/DefaultSearchAdminRequestExecutorService.java

```diff
@@ -0,0 +1,124 @@
+package com.valtech.aem.saas.core.http.client;
+
+import com.valtech.aem.saas.api.request.SearchRequest;
+import com.valtech.aem.saas.core.http.response.SearchResponse;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Synchronized;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.osgi.services.HttpClientBuilderFactory;
+import org.osgi.service.component.annotations.*;
+import org.osgi.service.component.propertytypes.ServiceDescription;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.AttributeType;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+
+import java.util.Optional;
+
+@Slf4j
+@Component(service = SearchAdminRequestExecutorService.class,
+           immediate = true)
+@ServiceDescription("Search as a Service - Search Admin Request Executor Service")
+@Designate(ocd = DefaultSearchAdminRequestExecutorService.Configuration.class)
+public class DefaultSearchAdminRequestExecutorService implements SearchAdminRequestExecutorService {
+
+    @Reference
+    private HttpClientBuilderFactory httpClientBuilderFactory;
+
+    @Getter(value = AccessLevel.PRIVATE,
+            onMethod_ = {@Synchronized})
```

Comment:
what is this annotation synchronized doing?

Reply:
It is similar to adding the **synchronized** modifier. According to lombok docs (https://projectlombok.org/features/Synchronized), it is a safer variant.